app/code/Magento/Ui/view/base/web/js/form/form.js

```diff
@@ -54,6 +54,11 @@ define([
                     }
                     break;
 
+                case 'select-multiple':
+                    var name = item.name.substring(0,(item.name.length - 2)); //remove [] from the name ending
+                    result[name] = Array.prototype.slice.call(item.selectedOptions).map(function(o) {return o.value;});
```

Comment:
Since you are not mutating `item.selectedOptions` in place, there should be no need to clone it with `Array.prototype.slice`. Also, It might be clearer to use Underscore here, because this pattern is very common:

```js
result[name] = _.pluck(item.selectedOptions, 'value');
```